pkg/iptables/iptables_test.go

```diff
@@ -169,6 +170,45 @@ func TestOutput(t *testing.T) {
 	}
 }
 
+func TestConcurrencyWithWait(t *testing.T) {
+	RunConcurrencyTest(t, true)
+}
+
+func TestConcurrencyNoWait(t *testing.T) {
```

Comment:
This test does indeed fail without the patch, although I tested it on a sufficiently new version of iptables where the failure case is practically deterministic because it still checks the lock.